hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/upgrade/DataNodeUpgradeFinalizer.java

```diff
@@ -41,63 +42,58 @@
     BasicUpgradeFinalizer<DatanodeStateMachine, HDDSLayoutVersionManager> {
   private static final Logger LOG =
       LoggerFactory.getLogger(DataNodeUpgradeFinalizer.class);
-  private DatanodeStateMachine datanodeStateMachine;
 
-  public DataNodeUpgradeFinalizer(HDDSLayoutVersionManager versionManager,
-                                  String optionalClientID) {
+  public DataNodeUpgradeFinalizer(HDDSLayoutVersionManager versionManager) {
     super(versionManager);
-    clientID = optionalClientID;
   }
 
   @Override
   public StatusAndMessages finalize(String upgradeClientID,
                                     DatanodeStateMachine dsm)
       throws IOException {
-    datanodeStateMachine = dsm;
-    StatusAndMessages response = preFinalize(upgradeClientID, dsm);
-    if (response.status() != FINALIZATION_REQUIRED) {
-      return response;
-    }
-    try {
-      getFinalizationExecutor().execute(dsm.getLayoutStorage(),
-          this);
-    } catch (Exception e) {
-      e.printStackTrace();
-      throw new IOException(e.getMessage());
-    }
-    return STARTING_MSG;
+    return super.finalize(upgradeClientID, dsm);
```

Comment:
after refactoring we don't need this method as it is just delegating to super.

Reply:
Fixed